File: test_weather_mqtt.py
#!/usr/bin/env python3

import paho.mqtt.client as mqtt
import json
import time

# MQTT Configuration
MQTT_BROKER = "192.168.0.199"
MQTT_PORT = 1883
MQTT_USERNAME = "alarmpanelgarage"
MQTT_PASSWORD = "alarmpanelgarage"
DEVICE_NAME = "rajofficemac"

def on_connect(client, userdata, flags, rc):
    print(f"Connected with result code {rc}")

def on_publish(client, userdata, mid):
    print(f"Message published successfully (mid: {mid})")

def send_weather_command():
    client = mqtt.Client()
    client.username_pw_set(MQTT_USERNAME, MQTT_PASSWORD)
    client.on_connect = on_connect
    client.on_publish = on_publish
    
    try:
        client.connect(MQTT_BROKER, MQTT_PORT, 60)
        client.loop_start()
        
        # Wait for connection
        time.sleep(2)
        
        # Weather widget command with demo API key (will show error but should create tile)
        weather_command = {
            "command": "open_weather_client",
            "api_key": "demo_key_123",
            "location": "London"
        }
        
        topic = f"kingkiosk/{DEVICE_NAME}/command"
        message = json.dumps(weather_command)
        
        print(f"Sending weather command to topic: {topic}")
        print(f"Command: {message}")
        
        result = client.publish(topic, message)
        
        # Wait for publish to complete
        time.sleep(2)
        
        client.loop_stop()
        client.disconnect()
        
        print("Weather command sent successfully!")
        print("Check the Flutter app - a weather tile should appear")
        
    except Exception as e:
        print(f"Error sending command: {e}")

if __name__ == "__main__":
    send_weather_command()

File: test_weather_mqtt.dart
import 'dart:convert';
import 'dart:io';
import 'package:mqtt_client/mqtt_client.dart';
import 'package:mqtt_client/mqtt_server_client.dart';

void main() async {
  // MQTT Configuration
  final String mqttServer = '192.168.0.199';
  final int mqttPort = 1883;
  final String username = 'alarmpanelgarage';
  final String password = 'alarmpanelgarage';
  final String topic = 'kingkiosk/rajofficemac/command';

  // Create MQTT client
  final client = MqttServerClient(mqttServer, 'weather_test_client');
  client.port = mqttPort;
  client.logging(on: true);
  client.keepAlivePeriod = 20;
  client.onDisconnected = onDisconnected;
  client.onConnected = onConnected;

  // Connect with credentials
  final connMessage = MqttConnectMessage()
      .withClientIdentifier('weather_test_client')
      .authenticateAs(username, password)
      .startClean()
      .withWillQos(MqttQos.atMostOnce);

  client.connectionMessage = connMessage;

  try {
    print('Connecting to MQTT broker at $mqttServer:$mqttPort...');
    await client.connect();
  } catch (e) {
    print('Error connecting to MQTT broker: $e');
    client.disconnect();
    exit(1);
  }

  if (client.connectionStatus!.state == MqttConnectionState.connected) {
    print('✅ Connected to MQTT broker successfully!');

    // Create weather widget command
    final weatherCommand = {
      'command': 'open_weather_client',
      'payload': {
        'window_id': 'weather_test_${DateTime.now().millisecondsSinceEpoch}',
        'window_name': 'Test Weather Widget',
        'api_key': 'YOUR_OPENWEATHER_API_KEY', // You'll need to replace this
        'units': 'metric',
        'language': 'en',
        'show_forecast': true,
        'auto_refresh': true,
        'refresh_interval': 300,
        // Optional: specify location, otherwise it will use device location
        // 'location': 'London,UK',
        // 'latitude': 51.5074,
        // 'longitude': -0.1278,
      }
    };

    final jsonCommand = jsonEncode(weatherCommand);
    print('📤 Sending weather command:');
    print(jsonCommand);

    // Publish the command
    final builder = MqttClientPayloadBuilder();
    builder.addString(jsonCommand);

    client.publishMessage(topic, MqttQos.atMostOnce, builder.payload!);
    print('✅ Weather command sent successfully!');

    // Wait a moment then disconnect
    await Future.delayed(Duration(seconds: 2));
    client.disconnect();
    print('👋 Disconnected from MQTT broker');
  } else {
    print('❌ Failed to connect to MQTT broker');
    client.disconnect();
    exit(1);
  }
}

void onConnected() {
  print('🔗 MQTT client connected');
}

void onDisconnected() {
  print('📤 MQTT client disconnected');
}
